.github/workflows/goreleaser-build.yaml

```diff
@@ -3,95 +3,87 @@ name: GoReleaser Build
-            echo ::set-output name=GCP_PROJECT::cyderes-dev
-            echo ::set-output name=PUSH_FLAG::
-            echo ::set-output name=APP_VERSION_TAG::latest
-            echo ::set-output name=GCP_ARGO_SERVICE_ACCOUNT::${{ secrets.GCP_ARGO_SERVICE_ACCOUNT_DEV }}
-          else
-            echo ::set-output name=GCP_PROJECT::cyderes-dev
-            echo ::set-output name=PUSH_FLAG::--no-push
-            echo ::set-output name=APP_VERSION_TAG::latest
-            echo ::set-output name=GCP_ARGO_SERVICE_ACCOUNT::${{ secrets.GCP_ARGO_SERVICE_ACCOUNT_DEV }}
-          fi
-
       - name: Docker login
-        uses: 'docker/login-action@v1'
-        if: ${{ github.event_name != 'release' }}
+        uses: docker/login-action@v2
         with:
           registry: ${{ inputs.ARTIFACT_REGISTRY_HOST }}
           username: _json_key_base64
-          password: ${{ steps.set_env.outputs.GCP_ARGO_SERVICE_ACCOUNT }}
-
-      - uses: actions/setup-go@v3
+          password: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+      - name: Setup Go
+        uses: actions/setup-go@v4
         with:
-          go-version: '^1.13.1'
-
+          go-version-file: ${{ inputs.GO_MOD_PATH }}
       - name: Get Go version
         id: go_version
         run: echo ::set-output name=GO_VERSION::$(go version | awk '{print $3}')
-
-      - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v3
+      - name: Run Goreleaser
+        uses: goreleaser/goreleaser-action@v4
         with:
           distribution: goreleaser
-          version: latest
-          args: release --snapshot --rm-dist --skip-sign --debug
+          version: ${{ inputs.GORELEASER_VERSION }}
+          args: ${{ inputs.GORELEASER_ARGS }}
         env:
+          GOPRIVATE: ${{ inputs.GOPRIVATE }}
+          CGO_ENABLED: ${{ inputs.CGO_ENABLED }}
           GITHUB_TOKEN: ${{ secrets.CI_GITHUB_TOKEN }}
           GOVERSION: ${{ steps.go_version.outputs.GO_VERSION }}
-          REGISTRY: ${{ inputs.ARTIFACT_REGISTRY_HOST }}/${{ steps.set_env.outputs.GCP_PROJECT }}/${{ inputs.ARTIFACT_REGISTRY_REPO }}
-
+          REGISTRY: ${{ inputs.ARTIFACT_REGISTRY_HOST }}/${{ inputss.GCP_PROJECT }}/${{ inputs.ARTIFACT_REGISTRY_REPO }}
       - name: Docker push
-        if: ${{ github.event_name == 'push' || github.event_name == 'release' }}
-        run: |
-          PROJECT_NAME=$(cat .goreleaser.yml | grep project_name | awk '{print $2}')
-          SOURCE_IMAGE=$(docker image ls | grep $PROJECT_NAME | grep ${{ inputs.APP_VERSION }} | awk '{print $1}')
-          TARGET_IMAGE="${{ inputs.ARTIFACT_REGISTRY_HOST }}/${{ steps.set_env.outputs.GCP_PROJECT }}/${{ inputs.ARTIFACT_REGISTRY_REPO }}/${PROJECT_NAME}:${{ inputs.APP_VERSION }}"
-          docker image tag ${SOURCE_IMAGE} ${TARGET_IMAGE}
-          docker push ${TARGET_IMAGE}
\ No newline at end of file
+        run: docker push ${REGISTRY}/${{ github.event.repository.name }}:${{ inputs.APPLICATION_VERSION }}
```

Comment:
do we not need/want to tag the image anymore?

Reply:
`${{ inputs.APPLICATION_VERSION }}` This is the tag, for dev we would just set this to `latest` and for prod it would be the git tag.